internal/controllers/utils/conditions.go

```diff
@@ -136,3 +138,14 @@ func IsSmoRegistrationCompleted(cr *inventoryv1alpha1.Inventory) bool {
 		string(InventoryConditionTypes.SmoRegistrationCompleted))
 	return condition != nil && condition.Status == metav1.ConditionTrue
 }
+
+// IsClusterUpgradeInProgress checks if the cluster upgrade condition status is in progress
+func IsClusterUpgradeInProgress(cr *provisioningv1alpha1.ProvisioningRequest) bool {
+	condition := meta.FindStatusCondition(cr.Status.Conditions, (string(PRconditionTypes.UpgradeCompleted)))
```

Comment:
This probably works in most cases. It might be a bit more reliable if we do this based on the existence of the IBGU CR.

Reply:
The essence of this check is more like "if there is any upgrade stuff that needs to be taken care of by the handleUpgrade function".